### DISCUSSION TITLE
feat: Detailed Agent reports and time tracking

### DISCUSSION THREAD

**Discussion by rahulmr** [chatwoot/chatwoot#8458] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
Need to track agent working hours but the api for agent only provides below information. 
Even the user details api does not provide information sign in count, sign in / sign out time for today 

**Describe the solution you'd like**
There is no way to monitor the agents information in detail. Agents reports are very primitive.
Only number of conversations and messages is shown and total resolution time is given but no details per conversation.

**Describe alternatives you've considered**
Only alternative seems to directly get information using rails.

**Additional context**
I want to monitor agents conversations as an admin and also want to check time spent by them on conversations or chatwoot platform in general.

**Comment by pranavrajs** [comment]
@rahulmr Thanks for creating the request. I would like to understand a little bit about the use-case. Are you trying to learn more about the productivity of an agent? Also, it would be great if you could share some of the benchmarks / KPIs you internally track for an agent.

**Comment by rahulmr** [comment]
@pranavrajs thanks for your reply. Yes you got it exactly, I have created a small vue page view to capture agent productivity. Currently, when I fetch agent details for a particular account I see only few details. Is there any way to capture more details like sign in, sign out, time spent per conversation per day, last sign in, last sign out, last active or availability, etc. I was also thinking of capturing timestamps where in we can get information when the agent status was changed say from active, busy to offline.
Was also thinking of few stuff where I can filter conversations based on agent and know how many conversations he is working on today.
I also found that there is no way to get conversations using since until filter but it only uses page as an option and fetch 25 conversations per page, suppose I want only 10 conversations per page (like I see in contact list page) instead of 25 (same logic is there in case of CSAT surveys which also has pagination) Since mine is self hosted one I can modify my rails and vue js code.
Please let me know if you can guide me on this approach.
I have already created the pages just need to fetch data properly.

**Comment by rahulmr** [comment]
@pranavrajs any insights on this? I had further discussion and came to know that we also want time spent per conversation per day by agent along with a date filter.

**Comment by rahulmr** [comment]
@pranavrajs @sojan-official guys any updates on this?

**Comment by sojan-official** [comment]
@rahulmr Thanks for the information. we will keep track of this and share further updates as we have them. 

If you are looking into solving this on your end 
- some of the information about the user like last sign in , sign out are available in the database even though not on the API. you can look into pulling that information out 
ref: https://github.com/chatwoot/chatwoot/blob/5ee209c079fedf8f76e2054dac568c58257b0d9a/app/models/user.rb#L16 

you also look into additional configurations available in https://github.com/heartcombo/devise 

- Conversation end point is controller by https://github.com/chatwoot/chatwoot/blob/5ee209c079fedf8f76e2054dac568c58257b0d9a/app/finders/conversation_finder.rb#L1 , you can look into adding pagination into it.  Also the newest version of chatwoot can conversation filter APIs which could be helpful for you to look for conversations handled by a specific agent

**Comment by rahulmr** [comment]
@sojan-official thanks a lot replying. I am already aware of `last_sign_in_at` but actually not able to find any information on something related to `sign_out` there is nothing like `last_sign_out_at`, I see only `sign_in_count` etc.
Seems like devise does not provide more information on sign_out or destroy session. Not sure how I can add sign_out time to any table.
